src/commonMain/kotlin/fr/acinq/lightning/payment/OutgoingPaymentPacket.kt

```diff
@@ -26,18 +26,13 @@ object OutgoingPaymentPacket {
     /**
      * Build an encrypted onion packet from onion payloads and node public keys.
      */
-    private fun buildOnion(nodes: List<PublicKey>, payloads: List<PaymentOnion.PerHopPayload>, associatedData: ByteVector32, payloadLength: Int): PacketAndSecrets {
+    private fun buildOnion(nodes: List<PublicKey>, payloads: List<PaymentOnion.PerHopPayload>, associatedData: ByteVector32, payloadLength: Int?): PacketAndSecrets {
         require(nodes.size == payloads.size)
         val sessionKey = Lightning.randomKey()
         val payloadsBin = payloads
-            .map {
-                when (it) {
-                    is PaymentOnion.ChannelRelayPayload -> it.write()
-                    is PaymentOnion.NodeRelayPayload -> it.write()
-                    is PaymentOnion.FinalPayload -> it.write()
-                }
-            }
-        return Sphinx.create(sessionKey, nodes, payloadsBin, associatedData, payloadLength)
+            .map { it.write() }
```

Comment:
nit: we don't really need to put this on a new line, it's more readable on the same line as `val payloadsBin = payloads`